main.js

```diff
@@ -35,6 +60,33 @@ function makeRequest(fn, uri, options, callback) {
     }
 }
 
+Trello.prototype.makeRequest = function (requestMethod, uri, options, callback) {
+    if (typeof options === 'undefined') {
+        options = {};
+    }
+    if (typeof requestMethod !== 'string') {
+        throw new TypeError("requestMethod should be a string");
+    }
+    if (typeof options !== 'object') {
+        throw new TypeError("options should be an object");
+    }
+
+    var method = requestMethod.toLowerCase();
+    var methods = {
+        'post': rest.post,
+        'get': rest.get,
+        'put': rest.put,
+        'delete': rest.del
+    };
+
+    if (Object.keys(methods).indexOf(method) < 0) {
+        throw new Error("Unsupported requestMethod. Pass one of these methods: POST, GET, PUT, DELETE.");
+    }
+    var keyTokenObj = this.createQuery();
+    var query = Object.assign(options, keyTokenObj);
```

Comment:
Shouldn't that be `Object.assign({}, options, keyTokenObj), otherwise its mutating the input argument `options`.
This might be unexpected by the caller of the function.